File: csharp/series/Series.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length || sliceLength <= 0)
        {
            throw new ArgumentException("Invalid slice length, should be equal or lesser than the numbers length, but also bigger than zero.", nameof(sliceLength));
        }

        if (string.IsNullOrWhiteSpace(numbers))
        {
            throw new ArgumentException("Numbers should not be empty or null.", nameof(numbers));
        }

        return numbers.Slice(sliceLength).ToArray();
    }

    public static IEnumerable<string> Slice(this string numbers, int sliceLength)
    {
        var index = 0;
        var limit = numbers.Length - sliceLength;

        while(index <= limit)
        {
            var slice = numbers.Substring(index, sliceLength);
            index++;

            yield return int.TryParse(slice, out _)
                ? slice
                : throw new ArgumentException("There should be only numbers inside the string.", nameof(numbers));
        }
    }
}

File: typescript/series/series.ts
export class Series {
  private readonly series: number[];

  constructor(series: string) {
    if (!series) {
      throw new Error('series cannot be empty');
    }

    this.series = series.split('').map(char => +char);
  }

  *generate(sliceLength: number): Iterable<number[]> {
    var index = 0;

    while (index + sliceLength <= this.series.length) {
      const endIndex = index + sliceLength;
      yield this.series.slice(index++, endIndex);
    }
  }

  slices(sliceLength: number): number[][] {
    if (sliceLength > this.series.length) {
      throw new Error('slice length cannot be greater than series length');
    }

    if (sliceLength == 0) {
      throw new Error('slice length cannot be zero');
    }

    if (sliceLength < 0) {
      throw new Error('slice length cannot be negative');
    }

    return Array.from(this.generate(sliceLength));
  }
}
